foundry/foundry.py

```diff
@@ -140,210 +140,126 @@ def __init__(
+        """Search available Foundry datasets
 
-    def load(self, name, download=True, globus=True, verbose=False, metadata=None, authorizers=None, **kwargs):
-        """Load the metadata for a Foundry dataset into the client
         Args:
-            name (str): Name of the foundry dataset
-            download (bool): If True, download the data associated with the package (default is True)
-            globus (bool): If True, download using Globus, otherwise https
-            verbose (bool): If True print additional debug information
-            metadata (dict): **For debug purposes.** A search result analog to prepopulate metadata.
+            query (str): query string to match
+            limit (int): maximum number of results to return
 
-        Keyword Args:
-            interval (int): How often to poll Globus to check if transfers are complete
-
-        Returns
-        -------
-            self
+        Returns:
+            List[FoundryDataset]: List of search results as FoundryDatset objects
         """
 
-        # handle empty dataset name (was returning all the datasets)
-        if not name:
-            raise ValueError("load: No dataset name is given")
-
-        if metadata:
-            res = metadata
-
-        # MDF specific logic
-        if is_doi(name) and not metadata:
-            res = self.forge_client.match_resource_types("dataset")
-            res = res.match_dois(name).search()
-
+        if (query is not None) and (is_doi(query)):
+            metadatas = [self.get_metadata_by_doi(query)]
         else:
-            res = self.forge_client.match_field(
-                "mdf.organizations", self.config.organization
-            ).match_resource_types("dataset")
-            res = res.match_field("mdf.source_id", name).search()
-
-        # unpack res, handle if empty
-        if len(res) == 0:
-            raise Exception(f"load: No metadata found for given dataset {name}")
+            metadatas = self.get_metadata_by_query(query, limit)
 
-        # if search returns multiple results, this automatically uses first result, while warning the user
-        if len(res) > 1:
-            warnings.warn("Multiple datasets found for the given search query. Using first dataset")
-        res = res[0]
+        if len(metadatas) == 0:
+            raise Exception(f"load: No results found for the query '{query}'")
 
-        try:
-            res["dataset"] = res["projects"][self.config.metadata_key]
-        except KeyError as e:
-            raise Exception(f"load: not able to index with metadata key {self.config.metadata_key}") from e
+        foundry_datasets = []
+        for metadata in metadatas:
+            ds = self.dataset_from_metadata(metadata)
+            if ds:
+                foundry_datasets.append(ds)
 
-        del res["projects"][self.config.metadata_key]
+        print(f"Search for '{query}' returned {len(foundry_datasets)} foundry datasets out of {len(metadatas)} matches")
```

Comment:
So I tried this with logger at the info level, but once I set the log level to INFO, I get a ton of spew from what I believe is the forge client: ![Screenshot 2023-11-10 at 10 47 30 AM](https://github.com/MLMI2-CSSI/foundry/assets/1233784/203f0162-5512-4f66-96d5-363254f1c90a)

I also considered a WARNING level, but that seemed inappropriate to use as it isn't really a warning. The only logical way I could envision communicating this back to the user (as horrendous as it is) was via a print statement. Do you have other thoughts about how this could be communicated?

The other part of this is if the user provides an arg to the `limit` kwarg, this at least lets them know that the limit was implemented, but only a portion of those returned were valid FoundryDatasets. Otherwise, they can specify a limit of e.g. 10 but only get 1 result (e.g. `f.search('Ben', limit = 10)`) without any explanation.

Reply:
oh man that's some terrible output. hmm I saw that you went with the WARNING implementation -- I guess I'll have to see what that looks like in practice. bummer that the log is so clogged :/